ivy/__init__.py

```diff
@@ -77,7 +77,12 @@ def __new__(cls, dev_str):
         if dev_str != "":
             ivy.utils.assertions.check_elem_in_list(dev_str[0:3], ["gpu", "tpu", "cpu"])
             if dev_str != "cpu":
-                # ivy.assertions.check_equal(dev_str[3], ":")
+                ivy.assertions.check_equal(
```

Comment:
Rather than enforcing the use of device_id, can we instead assume it to be `0` when not passed? We should then be raising a warning saying that the device id wasn't passed in the input string, so is being considered as `0`.